projects/ui/src/components/Swap/Actions/Swap.tsx

```diff
@@ -143,18 +149,46 @@ const SwapForm: FC<FormikProps<SwapFormValues> & {
   const shouldApprove = tokensMatch 
     /// If matching tokens, only approve if input token is using EXTERNAL balances.
     ? modeIn === FarmFromMode.EXTERNAL
-    /// Otherwise, approve if we expect to use an EXTERNAL baalnce.
+    /// Otherwise, approve if we expect to use an EXTERNAL balance.
     : (
       (expectedFromMode === FarmFromMode.EXTERNAL
       || expectedFromMode === FarmFromMode.INTERNAL_EXTERNAL)
     );
 
+  const buildSwapHelper = useCallback((uiTokenIn: Token, uiTokenOut: Token, farmFrom: FarmFromMode, farmTo: FarmToMode) => {
+    const sdkTokenIn = sdk.tokens.findByAddress(uiTokenIn.address);
+    if (!sdkTokenIn) {
+      throw new Error(`Address of ${uiTokenIn.symbol} was not found in SDK tokens.`);
+    }
+    const sdkTokenOut = sdk.tokens.findByAddress(uiTokenOut.address);
+    if (!sdkTokenOut) {
+      throw new Error(`Address of ${uiTokenOut.symbol} was not found in SDK tokens.`);
+    }
+
+    return sdk.swap.buildSwap(sdkTokenIn, sdkTokenOut, account!, farmFrom, farmTo);
+  }, [sdk.tokens, sdk.swap, account]);
```

Comment:
Having sdk.tokens and sdk.swap here is very strange. I know this is to make [react-hooks/exhaustive-deps] happy, but feels so wrong, lol